pysqream_blue/cursor.py

```diff
@@ -132,19 +141,18 @@ def fetchall(self, bad_args=False):
     def get_statement_id(self):
         return self.stmt_id
 
-    def _request_compile(self, query: str):
-        try:
-            response: qh_messages.CompileResponse = self._compile(query)
+    def _request_compile_inner(self, query):
+        response: qh_messages.CompileResponse = self._compile(query)
 
-            if response.HasField('error'):
-                self.logs.log_and_raise(OperationalError,
-                              f'Query: {query}. Error while attempting to compile the query.\n{response.error}')
+        if response.HasField('error'):
+            self.logs.log_and_raise(OperationalError,
+                          f'Query: {query}. Error while attempting to compile the query.\n{response.error}')
 
-            self.logs.message(f'Query id: {self.stmt_id}. The query was successfully compiled. query type: '
-                              f'{self.query_type}.', self.logs.info)
+        self.logs.message(f'Query id: {self.stmt_id}. The query was successfully compiled. query type: '
+                          f'{self.query_type}.', self.logs.info)
 
-        except grpc.RpcError as rpc_error:
-            self.logs.log_and_raise(ProgrammingError, f'Query: {query}. Error from grpc while attempting to compile the query.\n{rpc_error}')
+    def _request_compile(self, query: str):
+        self._retry_function_on_stream_error(self._request_compile_inner, "compile the query", query)
```

Comment:
you can send only "compile", and on the error log add "the query" after the function action.